### PR TITLE
Split trigonometric/hyperbolic routines from math.cc

### PR DESCRIPTION
**Pull request by asi1024**
Part of #7143.

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/cuda_device/hyperbolic.cu [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include "chainerx/cuda/cuda_runtime.h"
+#include "chainerx/cuda/cuda_set_device_scope.h"
+#include "chainerx/cuda/elementwise.cuh"
+#include "chainerx/cuda/kernel_regist.h"
+#include "chainerx/cuda/numeric.cuh"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/kernels/hyperbolic.h"
+#include "chainerx/numeric.h"
+#include "chainerx/scalar.h"
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/cuda_device/hyperbolic.cu]: Let's do these inclusions in the `kernel_regist.h`s. Maybe in a separate PR.

File: chainerx_cc/chainerx/routines/math.h [hunk 2]
```diff
@@ -117,30 +117,8 @@ Array IsFinite(const Array& x);
 
 Array Tanh(const Array& x);
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/math.h]: Isn't `Tanh` to be included in the "hyperbolic" routines and kernels?

File: chainerx_cc/chainerx/kernels/trigonometric.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <cstdint>
```
Comment (hvy) [inline chainerx_cc/chainerx/kernels/trigonometric.h]: Ditto.

File: chainerx_cc/chainerx/routines/trigonometric.cc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+#include "chainerx/backward_builder.h"
+#include "chainerx/backward_context.h"
+#include "chainerx/dtype.h"
+#include "chainerx/error.h"
+#include "chainerx/graph.h"
+#include "chainerx/kernels/trigonometric.h"
+#include "chainerx/routines/math.h"
+#include "chainerx/routines/routines_util.h"
+#include "chainerx/routines/type_util.h"
+#include "chainerx/scalar.h"
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/trigonometric.cc]: Ditto.

File: chainerx_cc/chainerx/routines/trigonometric.cc [hunk 5]
```diff
@@ -0,0 +1,188 @@
+#include "chainerx/routines/trigonometric.h"
+
+#include "chainerx/array.h"
+#include "chainerx/backprop_mode.h"
+#include "chainerx/backward_builder.h"
+#include "chainerx/backward_context.h"
+#include "chainerx/dtype.h"
+#include "chainerx/error.h"
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/trigonometric.cc]: Ditto.

File: chainerx_cc/chainerx/routines/trigonometric.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "chainerx/array.h"
+#include "chainerx/scalar.h"
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/trigonometric.h]: Ditto.

File: chainerx_cc/chainerx/routines/type_util.h [hunk 7]
```diff
@@ -93,4 +93,15 @@ Dtype ResultType(Container args) {
     return resolver.Resolve();
 }
 
+namespace internal {
+
+inline Dtype GetMathResultDtype(Dtype dtype) {
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/type_util.h]: A cosmetic but how about defining this function inside the `namespace internal { ... }` that already exists at the top?

File: chainerx_cc/chainerx/cuda/cuda_device/hyperbolic.cu (hunk 1 above)
Comment (asi1024) [reply chainerx_cc/chainerx/cuda/cuda_device/hyperbolic.cu]: I'll fix them in another PR.

File: chainerx_cc/chainerx/routines/math.h (hunk 2 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/math.h]: Moved to `hyperbolic.h`

File: chainerx_cc/chainerx/kernels/hyperbolic.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstdint>
+
+#include "chainerx/array.h"
+#include "chainerx/kernel.h"
+#include "chainerx/scalar.h"
```
[the remark this answers is not in the document]
Comment (asi1024) [reply chainerx_cc/chainerx/kernels/hyperbolic.h]: Fixed.

File: chainerx_cc/chainerx/kernels/hyperbolic.h [hunk 9]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstdint>
```
[the remark this answers is not in the document]
Comment (asi1024) [reply chainerx_cc/chainerx/kernels/hyperbolic.h]: Fixed.

File: chainerx_cc/chainerx/routines/type_util.h (hunk 7 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/type_util.h]: Fixed.

**Comment by hvy** [comment]
Jenkins, test this please.

**Pull request merged**